File: src/main/scala/org/xbery/artbeams/products/repository/ProductFilter.scala
package org.xbery.artbeams.products.repository

import org.xbery.artbeams.common.assets.repository.AssetFilter

/**
  * @author Radek Beran
  */
case class ProductFilter(
  override val id: Option[String],
  override val ids: Option[Seq[String]],
  override val createdBy: Option[String],
  slug: Option[String],
  title: Option[String]
) extends AssetFilter

object ProductFilter {
  lazy val Empty = ProductFilter(None, None, None, None, None)
}

File: src/main/kotlin/org/xbery/artbeams/products/repository/ProductFilter.kt
package org.xbery.artbeams.products.repository

import org.xbery.artbeams.common.assets.repository.AssetFilter

/**
 * @author Radek Beran
 */
data class ProductFilter(
    override val id: String?,
    override val ids: List<String>?,
    override val createdBy: String?,
    val slug: String?,
    val title: String?
) : AssetFilter {
    companion object {
        val Empty: ProductFilter = ProductFilter(null, null, null, null, null)
    }
}
